File: hackerrank/easy/0030-priyanka-and-toys/js/solution.mjs
// solution.mjs

export function toys(w) {
  if (!w.length) return 0;
  const arr = [...w].sort((a, b) => a - b);
  let containers = 1;
  let base = arr[0];

  for (const weight of arr) {
    if (weight > base + 4) {
      containers++;
      base = weight;
    }
  }
  return containers;
}

// --- demo run (sample) ---
console.log(toys([1, 2, 3, 21, 7, 12, 14, 21])); // 4

File: hackerrank/easy/0030-priyanka-and-toys/python/solution.py
## solution.py

from typing import List

def toys(w: List[int]) -> int:
    if not w:
        return 0
    arr = sorted(w)
    containers = 1
    base = arr[0]
    for weight in arr:
        if weight > base + 4:
            containers += 1
            base = weight
    return containers


# --- demo run (sample) ---
if __name__ == "__main__":
    print(toys([1, 2, 3, 21, 7, 12, 14, 21]))  # 4